src/vaults/BaseVaults.sol

```diff
@@ -366,7 +373,8 @@ abstract contract BaseSyncDepositVault is BaseVault {
     /// @inheritdoc IERC7575
     function mint(uint256 shares, address receiver) public returns (uint256 assets) {
         assets = syncDepositManager.mint(address(this), shares, receiver, msg.sender);
-        SafeTransferLib.safeTransferFrom(asset, msg.sender, syncDepositManager.escrow(), assets);
+        // NOTE: For security reasons, transfer must stay at end of call
+        SafeTransferLib.safeTransferFrom(asset, msg.sender, address(_poolEscrowProvider.escrow(poolId)), assets);
```

Comment:
Post mortem: Fixes https://github.com/Recon-Fuzz/centrifuge-review-protocol-v3/issues/36

Reply:
LG, swaps the order to perform the unsafer call after